README.md

```diff
@@ -21,29 +21,71 @@ If you aren't using npm in your project, you can include reactWasm using UMD bui
 
 ## Usage
 
+### Render props
+
 Once you have installed react-wasm, supposing a CommonJS environment, you can import and use it in this way:
 
 ```js
-import Wasm from 'react-wasm';
+import Wasm from "react-wasm";
 
 // supposing an "add.wasm" module that exports a single function "add"
 const ExampleComponent = () => (
   <Wasm url="/add.wasm">
     {({ loading, error, data }) => {
-      if (loading) return 'Loading...';
-      if (error) return 'An error has occurred';
+      if (loading) return "Loading...";
+      if (error) return "An error has occurred";
 
       const { module, instance } = data;
-      return (
-        <div>
-          1 + 2 = {instance.exports.add(1, 2)}
-        </div>
-      );
+      return <div>1 + 2 = {instance.exports.add(1, 2)}</div>;
     }}
   </Wasm>
 );
 ```
 
+### Higher Order Component
+
+It's also possible to use the library using the HoC approach by importing the named `withWasm` function:
+
+```js
+import { withWasm } from "react-wasm";
+
+// supposing an "add.wasm" module that exports a single function "add"
+const ExampleComponent = ({ loading, error, data }) => {
+  if (loading) return "Loading...";
+  if (error) return "An error has occurred";
+
+  const { module, instance } = data;
+  return <div>1 + 2 = {instance.exports.add(1, 2)}</div>;
+};
+
+// with a config object
+const withAdd = withWasm({ url: "/add.wasm " });
+const EnhancedExample = withAdd(ExampleComponent);
+
+const App = () => <EnhancedExample />;
+
+// with the "url" prop
+const EnhancedExample = withWasm()(ExampleComponent);
+
+const App = () => <EnhancedExample url="/add.wasm" />;
+```
+
+The second argument of the `withWasm` function is a props mapper. If you want to customize the information your child
+component will receive from the underlying `Wasm` component, you can do:
+
+```javascript
+const mapToChild = ({ loading, errors, data }) => ({
+  hasLoaded: !loading,
+  hasError: !!error,
+  hello: data.hello
```

Comment:
I think that here we can replace the `hello` prop with an `add` prop, just to maintain the consistency with the example and with the `data` object (we have to check that `data` exists to avoid null pointers):
`add: data && data.instance.add`

After that I'll be happy to merge and publish a new release.
I really thank you for the effort and time, I really appreciate it! 😄 
Thank you so much!

Reply:
Well, I think it's my turn to thank for such a good quality review 👌 . Have to have worked with you on this 😊